src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -440,8 +443,19 @@ data OutboundQ msg nid = forall self .
 
       -- | Signal we use to wake up blocked threads
     , qSignal :: Signal CtrlMsg
+
+      -- | Mutable classification of peers.
+    , qClassifyPeers :: MVar (Classification nid)
```

Comment:
As @avieth and I just discussed, the answer to "Is there ever a peer that we've classified, but which isn't subscribed to us? I think so but maybe I'm mistaken." is no: we will subscribe all statically known peers to the queue when we initialize it (this isn't done yet but will be soon).